### PR TITLE
Tabbify app manager multimedia

### PR DESCRIPTION
**Pull request by orangejenny**
Hackathon: refactor multimedia pages into a single page with tabs that load on demand. Also, B3 upgrade these pages.

Putting on staging for @dimagi/product to have a look at.

QA ticket: http://manage.dimagi.com/default.asp?217346

@proteusvacuum 

Old multimedia entry point:
screen shot 2016-02-16 at 12 21 33 pm

New multimedia entry point:
screen shot 2016-02-16 at 12 20 18 pm

Old reference checker:
screen shot 2016-02-16 at 12 21 39 pm

New reference checker:
screen shot 2016-02-16 at 12 20 25 pm

Old bulk upload:
screen shot 2016-02-16 at 12 21 50 pm

New bulk upload:
screen shot 2016-02-16 at 12 20 48 pm

### CODE REVIEW

File: corehq/apps/app_manager/urls.py [hunk 1]
```diff
@@ -12,6 +13,9 @@
     url(r'^languages/bulk_app_translations/upload/$', 'upload_bulk_app_translations', name='upload_bulk_app_translations'),
     url(r'^multimedia/$', 'view_app', name='app_multimedia'),
     url(r'^multimedia_ajax/$', 'multimedia_ajax', name='app_multimedia_ajax'),
+    url(r'^multimedia_ajax_download/$', 'multimedia_ajax_download', name='app_multimedia_ajax_download'),
+    url(r'^multimedia_ajax_upload/$', BulkUploadMultimediaView.as_view(), name='app_multimedia_ajax_upload'),
+    url(r'^multimedia_ajax_references/$', MultimediaReferencesView.as_view(), name='app_multimedia_ajax_references'),
```
Comment (dimagimon) [inline corehq/apps/app_manager/urls.py]: E501 line too long (117 > 115 characters)

File: corehq/apps/app_manager/views/multimedia.py [hunk 2]
```diff
@@ -35,7 +35,7 @@ def multimedia_list_download(request, domain, app_id):
 
 
 @require_deploy_apps
-def multimedia_ajax(request, domain, app_id, template='app_manager/partials/multimedia_ajax.html'):
+def multimedia_ajax(request, domain, app_id, template='app_manager/partials/multimedia_ajax.html', extra_context=None):
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/multimedia.py]: E501 line too long (119 > 115 characters)

File: corehq/apps/app_manager/urls.py (hunk 1 above)
Comment (amsagoff) [inline corehq/apps/app_manager/urls.py]: This is great!

Feel free to ignore if scope creep - but what do you think of getting rid
of the "(Opens in a new window)" next to the mm reference checker button?
(The text looks out of place and I haven't seen that as an indicator
anywhere else on HQ/the internet)

File: corehq/apps/hqmedia/templates/hqmedia/partials/references.html [hunk 3]
```diff
@@ -0,0 +1,96 @@
+{% load hq_shared_tags %}
+{% load i18n %}
+
+{% block js %}
+    <script src="{% static 'hqwebapp/js/main.js' %}"></script>
+    <!-- todo add the media player back in -->
```
Comment (proteusvacuum) [inline corehq/apps/hqmedia/templates/hqmedia/partials/references.html]: I know you didn't write this, but does this still hold true?

**Comment by proteusvacuum** [comment]
:heart: :heart_decoration: 
This looks fantastic! Nothing blocking from me.

Kicked the build, as failure looked unrelated.

**Comment by benrudolph** [comment]
(i think travis is broken)

**Comment by czue** [comment]
nice, this is awesome! #orangebeast

**Comment by proteusvacuum** [comment]
@benrudolph it just needed a little :boot: 
:ballot_box_with_check:

**Comment by snopoke** [comment]
so much better!

File: corehq/apps/hqmedia/templates/hqmedia/partials/references.html (hunk 3 above)
Comment (orangejenny) [inline corehq/apps/hqmedia/templates/hqmedia/partials/references.html]: I'm not sure...but the line is almost three years old, so yeah, let's say this TODO has expired.

**Comment by calellowitz** [comment]
@orangejenny i just resolved a merge conflict with master in order to deploy staging. you probably want to make sure i did that correctly before removing the `open for review` tag. i was moving around some less imports which felt safe but i do not have a solid enough understanding of less to know that it was defnitely ok

**Comment by orangejenny** [comment]
@calellowitz I just checked out staging, looks like it's fine. Thanks for dealing with the conflict.

**Comment by nickpell** [comment]
merged master again

File: corehq/apps/hqmedia/static/hqmedia/js/hqmedia.reference_controller.js [hunk 4]
```diff
@@ -1,12 +1,11 @@
 /*globals $, _, django */
 
-function MultimediaReferenceController (references, obj_map, totals) {
+function MultimediaReferenceController (references, obj_map) {
```
Comment (dimagimon) [inline corehq/apps/hqmedia/static/hqmedia/js/hqmedia.reference_controller.js]: 'MultimediaReferenceController' is defined but never used (no-unused-vars)

**Comment by dannyroberts** [comment]
(I pushed the last commit in order to fix a problem where the deploy page was blank for RemoteApps. You can review, but I don't think it should be a problem.)

**Comment by orangejenny** [comment]
@dannyroberts thanks!

**Comment by orangejenny** [comment]
This seems unlikely to survive case management onboarding changes. Closing :(

**Pull request closed without merging**